e2e-tests/resources/authd-msentraid/browser_login.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+import argparse
+import locale
+
+import gi, os, sys
+
+gi.require_version("Gtk", "3.0")
+gi.require_version("Gdk", "3.0")
+gi.require_version("WebKit2", "4.1")
+
+from gi.repository import Gdk, Gtk  # type: ignore # The interpreter usually does not recognize imports from gi.repository
+
+# The import will be resolved at runtime, which means that the directory structure will be something like:
+# test_run_dir/
+#   resources/
+#     authd/
+#       browser_window.py <- the dependency
+#     authd-msentraid/
+#       browser_login.py  <- this file
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "authd")))
+from browser_window import (
+    BrowserWindow,
+    ascii_string_to_key_events,
+    maybe_run_offscreen,
+)  # type: ignore # This is resolved at runtime
+
+from generate_totp import generate_totp # type: ignore # This is resolved at runtime
+
+SNAPSHOT_INDEX = 0
+
+
+def main():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("username")
+    parser.add_argument("password")
+    parser.add_argument("device_code")
+    parser.add_argument("totp_secret")
```

Comment:
This is not the `totp_secret` but rather the `totp_code`, no?

Also can be optional or we require it all the times? In such case just call it `--totp-code`, and adapt the code depending whether is None or not.

Reply:
No, it's the base32 encoded secret key